src/flask/app.py

```diff
@@ -84,132 +84,46 @@ def wrapper_func(self, *args, **kwargs):
+    #: Set to ``None`` to use the app's :class:`~flask.app.Flask.json_decoder`.
+    json_decoder = None
 
-    .. versionadded:: 1.0
-       The ``host_matching`` and ``static_host`` parameters were added.
+    #: The name of the package or module that this app belongs to. Do not
+    #: change this once it is set by the constructor.
+    import_name = None
 
-    .. versionadded:: 1.0
-       The ``subdomain_matching`` parameter was added. Subdomain
-       matching needs to be enabled manually now. Setting
-       :data:`SERVER_NAME` does not implicitly enable it.
+    #: Location of the template files to be added to the template lookup.
+    #: ``None`` if templates should not be added.
+    template_folder = None
 
-    :param import_name: the name of the application package
-    :param static_url_path: can be used to specify a different path for the
-                            static files on the web.  Defaults to the name
-                            of the `static_folder` folder.
-    :param static_folder: The folder with static files that is served at
-        ``static_url_path``. Relative to the application ``root_path``
-        or an absolute path. Defaults to ``'static'``.
-    :param static_host: the host to use when adding the static route.
-        Defaults to None. Required when using ``host_matching=True``
-        with a ``static_folder`` configured.
-    :param host_matching: set ``url_map.host_matching`` attribute.
-        Defaults to False.
-    :param subdomain_matching: consider the subdomain relative to
-        :data:`SERVER_NAME` when matching routes. Defaults to False.
-    :param template_folder: the folder that contains the templates that should
-                            be used by the application.  Defaults to
-                            ``'templates'`` folder in the root path of the
-                            application.
-    :param instance_path: An alternative instance path for the application.
-                          By default the folder ``'instance'`` next to the
-                          package or module is assumed to be the instance
-                          path.
-    :param instance_relative_config: if set to ``True`` relative filenames
-                                     for loading the config are assumed to
-                                     be relative to the instance path instead
-                                     of the application root.
-    :param root_path: The path to the root of the application files.
-        This should only be set manually when it can't be detected
-        automatically, such as for namespace packages.
-    """
+    #: Absolute path to the package on the filesystem. Used to look up
+    #: resources contained in the package.
+    root_path = None
 
-    #: The class that is used for request objects.  See :class:`~flask.Request`
-    #: for more information.
-    request_class = Request
+    # MARK: additional vars added from Flask.
```

Comment:
Most of these should probably stay in `Flask`, blueprints don't have things like a Jinja environment or config.

Reply:
Should some of the Skeleton functions in Blueprints be overridden?

My workflow has been like this:
```
1. Comment out / Delete the function declaration (ie. def before_request) in Blueprints.
2. pytest - See what error comes up (usually function or variable definition undefined in Blueprints).
3. Move that func/var from Flask to Skeleton.
4. Repeat [2-3] until reached an **AssertionError**.
5. Update Blueprints register to merge self's dict into app's dict.
```

For context, I moved `debug` to Skeleton when I removed `before_request` (this was the first function we refactored) from Blueprints because pytest gave me an error that it wasn't implemented. I moved all the Jinja functions because I saw that `debug` sets `self.jinja_env.auto_reload` (though the tests pass without moving the Jinja functions over).

Reply:
**Update:**
I experimented a bit with our old branch and found that the `debug` error doesn't show if I remove `@setupmethod` from the respective function in `Skeleton`. Would this be the right approach to take?

Also in this case, how would we add `@setupmethod` for `Flask`?

Reply:
`debug` is only a property of `Flask`, it shouldn't be in `Skeleton`. Same with at least the stuff related to the Jinja environment and Config.

Reply:
Thanks for clarifying! 
I realized that `Blueprint` doesn't have any `@setupmethod`, which is where my `debug` error was coming from. Hence, if `Skeleton` contains the "common" function of `Flask` without the `@setupmethod`, then the tests work as expected.

Since `Flask` still uses `@setupmethod`, should `Flask` override the respective `Skeleton` functions somewhat like this?
```
 @setupmethod
 def url_defaults(self, f):
        super().url_defaults(...)
```